File: java-maven/plugins/amqp/src/test/java/{{ .PackagePath }}/{{ .NameWord }}/cases/AmqpActions.java
package {{ .Package }}.{{ .NameWord }}.cases;

import io.gatling.javaapi.core.ChainBuilder;
import org.galaxio.gatling.amqp.javaapi.AmqpDsl;

import static io.gatling.javaapi.core.CoreDsl.exec;
import static org.galaxio.gatling.javaapi.SimulationConfig.getStringParam;

public final class AmqpActions {

    public static final ChainBuilder publishMessage =
            exec(AmqpDsl.amqp("Publish")
                    .publish()
                    .queueExchange(getStringParam("amqpQueue"))
                    .textMessage("Hello message")
                    .messageId("1")
                    .priority(0));

    private AmqpActions() {
    }
}

File: kotlin-gradle/plugins/amqp/src/gatling/kotlin/{{ .PackagePath }}/{{ .NameWord }}/cases/AmqpActions.kt
package {{ .Package }}.{{ .NameWord }}.cases

import io.gatling.javaapi.core.ChainBuilder
import io.gatling.javaapi.core.CoreDsl.exec
import org.galaxio.gatling.amqp.javaapi.AmqpDsl
import org.galaxio.gatling.javaapi.SimulationConfig.getStringParam

object AmqpActions {
    val publishMessage: ChainBuilder = exec(
        AmqpDsl.amqp("Publish")
            .publish()
            .queueExchange(getStringParam("amqpQueue"))
            .textMessage("Hello message")
            .messageId("1")
            .priority(0)
    )
}
